File: system-test/dotnet/SystemTests/Latest/ExternalSystemContractTests/Tax/BaseTaxContractTest.cs
using SystemTests.Latest.ExternalSystemContractTests.Base;

namespace SystemTests.Latest.ExternalSystemContractTests.Tax;

public abstract class BaseTaxContractTest : BaseExternalSystemContractTest
{
    [Fact]
    public async Task ShouldBeAbleToGetTaxRate()
    {
        (await Scenario()
            .Given().Country().WithCode("US").WithTaxRate(0.09m)
            .Then().Country("US"))
            .HasTaxRateIsPositive();
    }
}

File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/latest/contract/tax/BaseTaxContractTest.java
package com.mycompany.myshop.systemtest.latest.contract.tax;

import com.mycompany.myshop.systemtest.latest.contract.base.BaseExternalSystemContractTest;
import org.junit.jupiter.api.Test;

public abstract class BaseTaxContractTest extends BaseExternalSystemContractTest {
    @Test
    void shouldBeAbleToGetTaxRate() {
        scenario
                .given().country().withCode("US").withTaxRate(0.09)
                .then().country("US").hasTaxRateIsPositive();
    }
}
